library/std/src/os/unix/io/mod.rs

```diff
@@ -54,6 +54,16 @@
 //! Like boxes, `OwnedFd` values conceptually own the resource they point to,
 //! and free (close) it when they are dropped.
 //!
+//! ## Required file-descriptor flags
+//!
+//! On Unix platforms, file descriptors that don't have the `O_CLOEXEC` flag
+//! set may be implicitly leaked into spawned child processes. This can be
+//! violate I/O safety, when the a file descriptor held in one part of the code
+//! is not intended to be exposed to child processes, and a spawn in another
+//! part of the code unknowingly exposes it. Consequently, Rust code that
+//! produces new `OwnedFd` or `BorrowedFd` values must either set the
+//! `O_CLOEXEC` flag, or be marked `unsafe`.
```

Comment:
I feel like there's some way of putting this more precisely (something about invariants or the like?) but I don't know what it is. (The same applies to the Windows documentation, if any change is made.)